src/vs/editor/common/languages/defaultDocumentColorsComputer.ts

```diff
@@ -0,0 +1,151 @@
+	findMatches(regex: RegExp): RegExpMatchArray[];
+}
+
+function _parseCaptureGroups(captureGroups: IterableIterator<string>) {
+	const values = [];
+	for (const captureGroup of captureGroups) {
+		const parsedNumber = Number(captureGroup);
+		if (parsedNumber || parsedNumber === 0 && captureGroup.replace(/\s/g, '') !== '') {
+			values.push(parsedNumber);
+		}
+	}
+	return values;
+}
+
+function _toIColor(r: number, g: number, b: number, a: number): IColor {
+	return {
+		red: r / 255,
+		blue: b / 255,
+		green: g / 255,
+		alpha: a
+	};
+}
+
+function _findRange(model: IDocumentColorComputerTarget, match: RegExpMatchArray): IRange | undefined {
+	const index = match.index;
+	const length = match[0].length;
+	if (!index) {
+		return;
+	}
+	const startPosition = model.positionAt(index);
+	const endPosition = model.positionAt(index + length);
+	const range: IRange = {
+		startLineNumber: startPosition.lineNumber,
+		startColumn: startPosition.column,
+		endLineNumber: endPosition.lineNumber,
+		endColumn: endPosition.column
+	};
+	return range;
+}
+
+function _findHexColorInformation(range: IRange | undefined, hexValue: string) {
+	if (!range) {
+		return;
+	}
+	const parsedHexColor = Color.Format.CSS.parseHex(hexValue);
+	if (!parsedHexColor) {
+		return;
+	}
+	return {
+		range: range,
+		color: _toIColor(parsedHexColor.rgba.r, parsedHexColor.rgba.g, parsedHexColor.rgba.b, parsedHexColor.rgba.a)
+	};
+}
+
+function _findRGBColorInformation(range: IRange | undefined, matches: RegExpMatchArray[], isAlpha: boolean) {
+	if (!range || matches.length !== 1) {
+		return;
+	}
+	const match = matches[0]!;
+	const captureGroups = match.values();
+	const parsedRegex = _parseCaptureGroups(captureGroups);
+	return {
+		range: range,
+		color: _toIColor(parsedRegex[0], parsedRegex[1], parsedRegex[2], isAlpha ? parsedRegex[3] : 1)
+	};
+}
+
+function _findHSLColorInformation(range: IRange | undefined, matches: RegExpMatchArray[], isAlpha: boolean) {
+	if (!range || matches.length !== 1) {
+		return;
+	}
+	const match = matches[0]!;
+	const captureGroups = match.values();
+	const parsedRegex = _parseCaptureGroups(captureGroups);
+	const colorEquivalent = new Color(new HSLA(parsedRegex[0], parsedRegex[1] / 100, parsedRegex[2] / 100, isAlpha ? parsedRegex[3] : 1));
+	return {
+		range: range,
+		color: _toIColor(colorEquivalent.rgba.r, colorEquivalent.rgba.g, colorEquivalent.rgba.b, colorEquivalent.rgba.a)
+	};
+}
+
+function _findMatches(model: IDocumentColorComputerTarget | string, regex: RegExp): RegExpMatchArray[] {
+	if (typeof model === 'string') {
+		return [...model.matchAll(regex)];
+	} else {
+		return model.findMatches(regex);
+	}
+}
+
+function computeColors(model: IDocumentColorComputerTarget): IColorInformation[] {
+	const result: IColorInformation[] = [];
+	// Early validation for RGB and HSL
+	const initialValidationRegex = /(rgb|rgba|hsl|hsla)(\([0-9\s,.\%]*\))|(#)([A-Fa-f0-9]{6}(?![A-Fa-f0-9]))|(#)([A-Fa-f0-9]{8}(?![A-Fa-f0-9]))/gm;
```

Comment:
You could use `\b` before (rgb|rgba|hsl|hsla) to make sure that words like `wahsl` donot match.
Same after the hex value.